### DISCUSSION TITLE
Validation transalations for array in request

### DISCUSSION THREAD

**Discussion by moamahfouz** [laravel/framework#50794] [Q&A]
Hello,

I'm currently facing challenges with translating Laravel array validation messages.
In my validation logic, I have an array key named 'room_name' that may contain multiple data elements:

```
Validator::validate($requestData, [
    'room_name' => 'required|array',
    'room_name.*' => 'string',
]);
```

Each element in the 'room_name' array corresponds to a room name input field in the HTML form:

```
<td><input type="text" class="form-control" value="{{$row}}" name="room_name[]"></td>
```

I've already added a translation for 'room_name' in the validation language file:

```
'room_name' => 'اسم القاعة او المعمل',
```

However, this translation only works for the first array element. Is there a way to translate all 'room_name' elements (e.g., room_name.1, room_name.2, ..., room_name.100) with the same translation?

Thank you.

**Comment by janzcio** [comment]
Hi! 

This is my idea.

In your validation language file (resources/lang/en/validation.php for English), add a wildcard message for array validation:

'custom' => [
    'room_name.*' => [
        'string' => 'The :attribute in position :array_position must be a string.', 
    ],
],

In your validation logic, you can pass the custom attribute names using the attributes() method:

Validator::validate($requestData, [
    'room_name' => 'required|array',
    'room_name.*' => 'string',
], [
    'room_name.*.string' => __('validation.custom.room_name.*.string', ['attribute' => __('validation.attributes.room_name')]),
])->setAttributeNames([
    'room_name.*' => __('validation.attributes.room_name'),
]);

Make sure to define the attribute name in your validation language file as well:
'attributes' => [
    'room_name' => 'Room Name',
],

Also, it will be best if you do this in laravel http request.
